lib/shared/src/chat/context-filter.ts

```diff
@@ -0,0 +1,64 @@
+export const CodyIgnoreFileName = '.codyignore'
+
+let codyIgnoreList: string[] = []
+
+/**
+ * Checks if the given file name should be ignored by Cody.
+ *
+ * @param fileName - The file name to check.
+ * @returns A boolean indicating if the file should be ignored.
+ */
+export function isCodyIgnoreFile(fileName?: string): boolean {
```

Comment:
Nit but the file in question is not the `codyignore` file but an ignore**d** file 🙈 

Also let's use URIs instead so it will cover edge cases 

```suggestion
export function isCodyIgnoredFile(uri?: vscode.Uri): boolean {
```